app/Jobs/ProductUpdateNotificationJob.php

```
@@ -0,0 +1,72 @@
+            $query->where('order_status', 'completed')
+                ->whereHas('items', function ($itemQuery) {
+                    $itemQuery->where('product_id', $this->product->id);
+                });
+        })->whereHas('notificationPreferences', function ($prefQuery) {
+            $prefQuery->where('product_id', $this->product->id)
+                ->where('type', 'product_update')
+                ->where('active', true);
+        })->get();
+
+        if ($usersToNotify->isEmpty()) {
+            Log::info('No users to notify for product update: ' . $this->product->name);
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'handle'. (Clean Code Rules)

(StaticAccess)